code/game/objects/items.dm

```diff
@@ -634,12 +641,30 @@
 		//START MONKEY
 		var/mob/living/carbon/monkey/MO = M
 		switch(slot)
+			if(slot_head)
+				if(MO.hat)
+					return 0
+				if( !(slot_flags & SLOT_HEAD) )
```

Comment:
Is that functionally equivalent? 00010 ^ 00010 = 00000, but !(00010 & 00010) = 11101